lib/OpenQA/BuildResults.pm

```diff
@@ -69,9 +69,9 @@ sub count_job ($job, $jr, $labels) {
 
 sub add_review_badge ($build_res) {
 
-    $build_res->{all_passed} = $build_res->{passed} + $build_res->{softfailed} >= $build_res->{total};
-    $build_res->{reviewed} = $build_res->{labeled} >= $build_res->{failed};
-    $build_res->{commented} = $build_res->{comments} >= $build_res->{failed};
+    $build_res->{all_passed} = $build_res->{passed} + $build_res->{softfailed} >= $build_res->{total} ? 1 : 0;
```

Comment:
I think you want

```suggestion
    $build_res->{all_passed} = !!($build_res->{passed} + $build_res->{softfailed} >= $build_res->{total} ? 1 : 0);
```

And accordingly for the other lines.

Reply:
I do? What does that do?

Reply:
per https://stackoverflow.com/questions/2168406/what-does-double-exclamation-point-mean that seems to be redundant, but less good (it says of the `!!` version "Actually it returns an empty string which numifys to 0", which sounds like the thing we're trying to avoid here). `? 1 : 0` is already a ternary that attempts to ensure we get 1 if the check passes and 0 if it fails; that's the main point of this PR. I don't see what wrapping it in a `!!` does.

Reply:
Sorry, I meant `!!($build_res->{passed} + $build_res->{softfailed} >= $build_res->{total})` as you wrote. The good thing is that we now clearly see the issue that you are trying to fix, e.g. https://build.opensuse.org/package/live_build_log/devel:openQA:GitHub:os-autoinst:openQA:PR-5735/openQA/openSUSE_Tumbleweed/x86_64
So I suggest we try with the numifying approach

Reply:
Yeah, as long as Cpanel::JSON::XS does return blessed objects as the default (or Mojolicious doesn't call it with different options), only numifying will help.